baseapp/query_router_test.go

```diff
@@ -0,0 +1,33 @@
+package baseapp
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+var testQuerier = func(_ sdk.Context, _ []string, _ *abci.RequestQuery) ([]byte, error) {
+	return nil, nil
+}
+
+func TestQueryRouter(t *testing.T) {
+	qr := NewQueryRouter()
+
+	// require panic on invalid route
+	require.Panics(t, func() {
+		qr.AddRoute("*", testQuerier)
+	})
+
+	qr.AddRoute("testRoute", testQuerier)
+	q := qr.Route("testRoute")
+	require.NotNil(t, q)
+
+	// require panic on duplicate route
+	require.Panics(t, func() {
+		qr.AddRoute("testRoute", testQuerier)
+	})
+}
```

Comment:
[optional] this can add a test case that asserts that a path with a `/` returns a panic.

Reply:
will do in a follow up, to not clutter git history here (if merged with a merged commit)